packages/SwingSet/test/vo-test-harness/test-vo-test-harness.js

```diff
@@ -0,0 +1,52 @@
+import { test, runVOTest } from '../../tools/vo-test-harness.js';
+
+async function voTestTest(t, mode) {
+  let makeThing;
+
+  // eslint thinks `retainer` is unused, but eslint is very wrong
+  // eslint-disable-next-line no-unused-vars
+  let retainer;
+
+  function prepare(vatData) {
+    makeThing = vatData.defineKind('thing', tag => ({ tag }), {
+      getTag: ({ state }) => state.tag,
+    });
+  }
+
+  function makeTestThing() {
+    const result = makeThing('arbitrary test thing');
+    if (mode === 'hold') {
+      retainer = result;
+    }
+    return result;
+  }
+
+  function testTestThing(thing, phase) {
+    if (phase === mode) {
+      t.fail(`deliberate failure in ${phase} phase`);
+    }
+    t.is(
+      thing.getTag(),
+      'arbitrary test thing',
+      `thing test failed in ${phase} phase`,
+    );
+  }
+
+  await runVOTest(t, prepare, makeTestThing, testTestThing);
+}
+
+test.failing('fail during "before" phase', async t => {
+  await voTestTest(t, 'before');
+});
```

Comment:
Copied from https://github.com/Agoric/agoric-sdk/pull/5224#discussion_r858149318

> Not relevant in this case.

Why?

Reply:
Because this is a test testing another test. The event being detected is not an exception being thrown, the event is a test failing as expected because it (successfully!) found the failure condition it was checking for.